### PR TITLE
Reduce memory usage

### PR DESCRIPTION
**Pull request by dmage**
Reduce notificationworker's memory usage from 110 MB to 75 MB.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -70,62 +59,34 @@
...
-
-    logger.debug("Loading test config.")
-    app.config.from_object(TestConfig())
-else:
-    from config import DefaultConfig
-
-    logger.debug("Loading default config.")
-    app.config.from_object(DefaultConfig())
+if not is_testing:
     app.teardown_request(database.close_db_filter)
```
Comment (dmage) [inline app.py:84]: Any ideas why we don't need it in testing?

File: util/ssl.py [hunk 2]
```diff
@@ -0,0 +1 @@
+SSL_FILENAMES = ["ssl.cert", "ssl.key"]
```
Comment (dmage) [inline util/ssl.py:1]: After the great cleanup (#1384) `SSL_FILENAMES` will be only in `notifications/notificationmethod.py`.

**Review by HammerMeetNail — APPROVED** [review]
I like the use of singletons, adds a nice structure and makes it clear which things should only have one instance.

File: singletons/avatar.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from avatars.avatars import Avatar
+from singletons.app import _app
```
[the remark this answers is not in the document]
Comment (flavianmissi) [reply singletons/avatar.py:2]: rather `from flask import current_app as app` https://flask.palletsprojects.com/en/1.1.x/api/#flask.current_app

File: workers/notificationworker/notificationworker.py [hunk 4]
```diff
@@ -48,15 +49,18 @@ def create_gunicorn_worker():
 
     utilizing this method will enforce a 1:1 quay worker to gunicorn worker ratio.
     """
+    from flask import Flask
+
     note_worker = NotificationWorker(
         notification_queue, poll_period_seconds=10, reservation_seconds=30, retry_after_seconds=30
     )
+    app = Flask(__name__)
```
Comment (flavianmissi) [inline workers/notificationworker/notificationworker.py:57]: hmmm `create_gunicorn_worker` creates this flask app, and `from singletons.config import app_config` creates another one - do we really need both? furthermore, what kind of side effects could this cause? 🤔

**Review by flavianmissi — COMMENTED** [review]
Left some comments, but looks good overall.

Do you think it's time we consider using an app factory? https://flask.palletsprojects.com/en/1.1.x/patterns/appfactories/

File: singletons/avatar.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmage) [reply singletons/avatar.py:2]: `_app` _is_ our Flask object. You can write:

```
from singletons.app import _app as app1
from app import app as app2
print(app1 is app2)
```

and it'll print you `True`. `current_app` is a smart proxy object that afaik only works in the context of an HTTP handler.

File: workers/notificationworker/notificationworker.py (hunk 4 above)
Comment (dmage) [reply workers/notificationworker/notificationworker.py:57]: This is just to make gunicorn happy. On next iterations most likely I'll remove `app` from GunicornWorker and replace it with some basic stub like `start_response("200 OK"); return []`. We don't expect it to serve any HTTP requests anyway, we only want it to watch code changes in local-dev.

**Comment by dmage** [comment]
App factory is a good way to eliminate global variables, maybe in 2023/2024 😄

**Pull request merged**